mne/decoding/receptive_field.py

```diff
@@ -0,0 +1,209 @@
+import numpy as np
+import matplotlib.pyplot as plt
+from sklearn.linear_model import Ridge
+from .base import get_coefs
+
+
+class ReceptiveField(object):
+    """Fit a receptive field model.
+
+    Parameters
+    ----------
+    stim_features : array, shape (n_features,)
+        Names for the input features to the model.
+    lags : array, shape (n_lags)
+        Time lags to be included in the model. This will be combined
+        with `sfreq` to convert these values into indices in the data.
+        If `sfreq` == 1, then these should be indices. If not, then
+        time lags should be in seconds.
+    sfreq : float
+        The sampling frequency of the data.
+    model : instance of sklearn model | None
+        The model used in fitting inputs and outputs. This can be any
+        sklearn-style model that contains a fit and predict method.
+    """
+    def __init__(self, stim_features, lags, sfreq=1., model=None):
+        self.stim_features = stim_features
+        self.lags = lags
+        self.model = Ridge() if model is None else model
+        self.sfreq = sfreq
+
+    def fit(self, X, y):
+        """Fit a receptive field model.
+
+        Parameters
+        ----------
+        X : array, shape ([n_epochs], n_channels, n_times)
+            The input features for the model.
```

Comment:
hmm, I think this should be hashed out before making this API change. I agree w you that the current structure is not sklearn compatible. On the other hand (at least in my case) not supporting epochs (or the ability to only fit the model on pre-defined timepoints) would make this class *much* less useful. It would mean that it's only suitable to fit on data where the stimuli are constant and continuous, which is not true for a trials-based setup (which my guess is many of the encoding model paradigms out there). One way to support fitting on subsets of time, albeit in a hackier way, would be to allow for a `mask_fit` parameter like we talked about implementing before. Then it'd do the delaying and would then pull out the subset of datapoints before calling `model.fit`.